src/main/java/com/github/dikhan/pagerduty/client/events/domain/Payload.java

```diff
@@ -170,7 +169,7 @@ public Builder setEventClass(String eventClass) {
          * @param customDetails An arbitrary JSON object containing any data you'd like included in the incident log.
          * @return Payload Builder with customDetails field populated to be able to keep populating the instance
          */
-        public Builder setCustomDetails(String customDetails) {
+        public Builder setCustomDetails(JSONObject customDetails) {
```

Comment:
The problem with having customDetails as String is that the JacksonObjectMapper would parse the value field enclosed in quotations, which would not be parsed nicely by the PagerDuty site.